dotnet/src/Planners/Planners.Handlebars/Handlebars/Extensions/KernelParameterMetadataExtensions.cs

```diff
@@ -73,7 +73,10 @@ private static void AddNestedComplexTypes(this HashSet<HandlebarsParameterTypeMe
         // Add nested complex types
         foreach (var property in properties)
         {
-            parameterTypes.UnionWith(property.PropertyType.ToHandlebarsParameterTypeMetadata());
+            if (!parameterTypes.Any(pt => pt.Name == property.PropertyType.Name))
```

Comment:
Shouldn't we match the types directly instead of name?
pt == property.PropertyType ?
```suggestion
            if (!parameterTypes.Any(pt => pt == property.PropertyType))
```

Reply:
pt isn't a Type, it's a `HandlebarsParameterTypeMetadata` which unfortunately doesn't have a Type in it. I'll see if it 's reasonable to add it...

Reply:
All good, if this will take long we can go with current state.